appOPHD/States/MapViewState.cpp

```diff
@@ -1039,6 +1042,28 @@ void MapViewState::placeRobodozer(Tile& tile)
 }
 
 
+namespace
+{
+	constexpr std::size_t AdjacentTileCount = 8;
+
+
+	std::array<bool, AdjacentTileCount> gatherExcavatedNeighbors(const TileMap& tileMap, const Tile& tile)
```

Comment:
Assuming we get rid of `mDirection`, we shouldn't need to gather an array of data for placement checks. We only need to know if there is some adjacent tile which is excavated and unoccupied.